src/submodules/io/jld2_integration.jl

```diff
@@ -0,0 +1,199 @@
+end
+
+to_serializable(t::ABM) = SerializableABM(
+    collect(values(t.agents)),
+    to_serializable(t.space),
+    to_serializable(t.properties),
+    t.rng,
+    t.maxid.x,
+)
+
+function to_serializable(t::GridSpace{D,P,W}) where {D,P,W}
+    pathfinder = to_serializable(t.pathfinder)
+    SerializableGridSpace{D,P,typeof(pathfinder)}(
+        size(t.s),
+        t.metric,
+        [(k, v) for (k, v) in t.hoods],
+        [(k, v) for (k, v) in t.hoods_tuple],
+        pathfinder,
+    )
+end
+
+to_serializable(t::ContinuousSpace{D,P,T}) where {D,P,T} =
+    SerializableContinuousSpace{D,P,T}(
+        to_serializable(t.grid),
+        t.dims,
+        t.spacing,
+        t.extent,
+    )
+
+to_serializable(t::GraphSpace{G}) where {G} = SerializableGraphSpace{G}(t.graph)
+
+to_serializable(t::Pathfinding.AStar{D,P,M}) where {D,P,M} =
+    SerializableAStar{D,P,M}(
+        [(k, collect(v)) for (k, v) in t.agent_paths],
+        t.grid_dims,
+        map(Tuple, t.neighborhood),
+        t.admissibility,
+        t.walkable,
+        t.cost_metric,
+    )
+
+function from_serializable(t::SerializableABM{S,A}; kwargs...) where {S,A}
+    abm = ABM(
+        A,
+        from_serializable(t.space; kwargs...);
+        scheduler = get(kwargs, :scheduler, Schedulers.fastest),
+        properties = from_serializable(t.properties; kwargs...),
+        rng = t.rng,
+    )
+    abm.maxid[] = t.maxid
+    for a in t.agents
+        add_agent_pos!(a, abm)
+    end
+    return abm
+end
+
+function from_serializable(t::SerializableGridSpace{D,P,W}; kwargs...) where {D,P,W}
+    s = Array{Vector{Int},D}(undef, t.dims)
+    for i in eachindex(s)
+        s[i] = Int[]
+    end
+    pathfinder = from_serializable(t.pathfinder; kwargs...)
+    return GridSpace{D,P,typeof(pathfinder)}(
+        s,
+        t.metric,
+        Dict(k => v for (k, v) in t.hoods),
+        Dict(k => v for (k, v) in t.hoods_tuple),
+        pathfinder,
+    )
+end
+
+function from_serializable(t::SerializableContinuousSpace{D,P,T}; kwargs...) where {D,P,T}
+    update_vel! = get(kwargs, :update_vel!, Agents.defvel)
+    ContinuousSpace(
+        from_serializable(t.grid; kwargs...),
+        update_vel!,
+        t.dims,
+        t.spacing,
+        t.extent,
+    )
+end
+
+from_serializable(t::SerializableGraphSpace; kwargs...) = GraphSpace(t.graph)
+
+from_serializable(t::SerializableAStar{D,P,M}; kwargs...) where {D,P,M} =
+    Pathfinding.AStar{D,P,M}(
+        Dict{Int,Pathfinding.Path{D}}(
+            k => Pathfinding.Path{D}(v...) for (k, v) in t.agent_paths
+        ),
+        t.grid_dims,
+        map(CartesianIndex, t.neighborhood),
+        t.admissibility,
+        t.walkable,
+        t.cost_metric,
+    )
+
+"""
+    AgentsIO.save_checkpoint(filename, model::ABM)
+
+Write the entire `model` to the JLD2 file specified by `filename`. Agent data, including
+multi-agent models, is also saved. Serialization capability depends on JLD2.jl.
```

Comment:
> including multi-agent models

Drop this aspect of the sentence. Sure, we need to account for that, but a user should not need to care if they have one or more agent types.

Reply:
> Serialization capability depends on JLD2.jl

This is also implied by the sentence above and considering we include JLD2 as a dependency inside Agents, this is only implicitly true, not explicitly (you don't have to `using Agents, JLD2` to get this to work`.